sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -395,12 +434,11 @@ object GeneratorNestedColumnAliasing {
             val updatedGenerate = rewrittenG.copy(generatorOutput = updatedGeneratorOutput)
 
             // Replace nested column accessor with generator output.
-            val attrExprIdsOnGenerator = attrToExtractValuesOnGenerator.keys.map(_.exprId).toSet
+            val generateOutputAttrExprId = attrToExtractValuesOnGenerator.keys.map(_.exprId).head
+            val generateOutputAttr = updatedGenerate.output
+              .find(_.exprId == generateOutputAttrExprId).get
```

Comment:
I changed the .getOrElse to .get to make it clearer - we should never fail to find the generator output attribute, if we want to be safer, we can use a require to check and return here, but should not use getOrElse because IMO if we run into the "else" branch it means an error.